server/src/main/java/com/ncc/neon/services/ClusterService.java

```diff
@@ -23,16 +26,190 @@
     /**
      * Clusters the results from the query.
      *
-     * TODO: Currently returns the same results i.e. method is a passthrough right now
-     *
      * @param tabularQueryResult the result of the unclustered query
      * @return the new clustered results
      */
     public TabularQueryResult cluster(TabularQueryResult tabularQueryResult) {
+        if (this.clusterClause.getFieldType().equals("number")) {
+            return this.aggregateNumber(tabularQueryResult);
+        }
```

Comment:
We might want a try/catch around this in case some of our assumptions (clusterClause.clusters aren't actually numbers, tabularQueryResult records don't actually have numbers, etc.) are invalid.

Reply:
Fixed.